bindings/python/src/session.cpp

```diff
@@ -206,68 +206,123 @@ namespace
+            {
+                p.save_path = extract<std::string>(params[key]);
+                continue;
+            }
 #ifndef TORRENT_NO_DEPRECATE
-        if (params.has_key("uuid"))
-            p.uuid = extract<std::string>(params["uuid"]);
+            else if(key == "resume_data")
+            {
+                std::string resume = extract<std::string>(params[key]);
+                p.resume_data.assign(resume.begin(), resume.end());
+                continue;
+            }
+            else if(key == "uuid")
+            {
+                p.uuid = extract<std::string>(params["uuid"]);
+                continue;
+            }
 #endif
-
-        if (params.has_key("file_priorities"))
-        {
-            list l = extract<list>(params["file_priorities"]);
-            int const n = int(boost::python::len(l));
-            for(int i = 0; i < n; i++)
-                p.file_priorities.push_back(extract<std::uint8_t>(l[i]));
-            p.file_priorities.clear();
+            else if(key == "storage_mode")
+            {
+                p.storage_mode = extract<storage_mode_t>(params[key]);
+                continue;
+            }
+            else if(key == "trackers")
+            {
+                p.trackers = extract<std::vector<std::string>>(params[key]);
+                continue;
+            }
+            else if(key == "url_seeds")
+            {
+                p.url_seeds = extract<std::vector<std::string>>(params[key]);
+                continue;
+            }
+            else if(key == "http_seeds")
+            {
+                // TODO: is it possible to create map["http_seeds"]=add_torrent_params::http_seeds?
+                // I don't how how to do it in c++ because we don't have reflections
+                // but I would like a map instead of if else and decltype does help us
+                p.http_seeds =
+                    extract<decltype(add_torrent_params::http_seeds)>(params[key]);
+                continue;
+            }
+            else if(key == "dht_nodes")
+            {
+                // TODO: did not find converter
+                //p.dht_nodes =
+                //    extract<std::vector<std::pair<std::string, int>>>(params[key]);
+                list l = extract<list>(params[key]);
+                int const n = int(boost::python::len(l));
+                for(int i = 0; i < n; i++)
+                    p.dht_nodes.push_back(
+                        // TODO: did not find converter either
+                        extract<std::pair<std::string, int>>(l[i]));
+                continue;
+            }
+            else if(key == "banned_peers")
+            {
+                // TODO: maybe the conversion of main or sub type of
+                // boost::asio::ip::basic_endpoint<boost::asio::ip::tcp>
+                // is missing because the other vector<type> definitions do work
+                p.banned_peers = extract<std::vector<lt::tcp::endpoint>>(params[key]);
```

Comment:
"peers" store the peer list from the last session. But I suppose when they're restored from resume data they won't be coming back this path, they will end up being passed to the function that takes an ``add_torrent_params``